Automattic/wp-calypso
# Popover: remove the rootClassName prop

Removes the `rootClassName` prop from `Popover` and related components. The prop doesn't have any useful usage and it styles a wrapper `div` that will disappear after the `RootChild` component is migrated to `React.createPortal` in #32899.

Except a lot of code that drill the props from parent components to children, there was only one actual usage: adding an unused `blog-stickers__popover` CSS class to a Reader popover.

This PR also avoids passing props to `RootChild` in `Popover` that are applied to a wrapper `div`, and passes them to the actual popover's inner `div` instead.

**How to test:**
Verify that `Popover` div markup looks reasonable, and all the a11y improvements that @ItsJonQ did in #32849 continue to work.

@ItsJonQ There's one thing I don't understand: your patch from #32849 adds a `tabIndex` to a non-interactive `div` element if the `isFocusOnShow` prop is enabled. But then the `eslint-plugin-jsx-a11y` ESLint rules complain:

> [`tabIndex` should only be declared on interactive elements](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-tabindex.md)

So, who is right here? What's the point of focusing a popover element that's not interactive?

When the `tabIndex` is there, I can see that when opening a popover (during domain search), the popover button loses focus and the popover element indeed becomes the active element (`document.activeElement`). But it doesn't show a focus ring or anything like that. Is it important for assistive tech like VoiceOver?
